src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -167,6 +172,11 @@ public class GenerateProtoTask extends DefaultTask {
     this.fileResolver = fileResolver
   }
 
+  void setExecCustomizer(Action<ExecSpec> execCustomizer) {
```

Comment:
The `ExecSpec` is way too powerful. It allows changing the whole command line. Users to shoot their feet in various ways. What use case do you want to support?

Reply:
My main use case is being able to set `PATH`, more specifically to make nodejs available via a downloaded binary that is not available in the system (for reproducible builds). Hacking around it is pretty painful

https://github.com/curioswitch/curiostack/blob/master/tools/gradle-plugins/gradle-curiostack-plugin/src/main/java/org/curioswitch/gradle/plugins/grpcapi/GrpcApiPlugin.java#L80

While having access only to environment might work, I can imagine plenty of use cases for wrapping the command line in another command that for example invokes a shell script to set variables up. Given protoc allows plugins that may have esoteric requirements, giving protoc invokers full control seems reasonable to me, especially since it requires opting in with an explicit method call. Dev tools should be flexible given the variety of dev envs out there.

Reply:
Would it be possible to create a wrapper script that invokes your plugin in the correct environment, and use that script's path as the plugin here?

Reply:
When downloading tools dynamically (for reproducible builds), it can be impractical to have a static script. Dynamically generating scripts is of course possible, like in that example, but it makes things pretty complicated, especially cross-platform.

Another example is this one, where it is useful to allow protobuf-gradle-plugin to download `protoc` from maven, but use a different executable `protowrap` for the actual execution. It would be pretty trivial by being able to wrap the `ExecSpec.commandLine`

https://github.com/infostellarinc/stellarstation-api/blob/master/stubs/golang/build.gradle#L105

I think when writing build tools, you often run into some weird cases and allowing customizability can be very important to users (more so than writing a core library) and have seen the pattern of allowing customization of exec in a few other plugins. But if it's too much API for this plugin, let's go ahead and close this PR.